### PR TITLE
Start splitting large C-files into smaller ones

### PR DESCRIPTION
**Pull request by sjoelund**
06inz.c is now split into parts of 2000 (configurable by
--equationsPerFile) number of equations per part (C-file).

Once implemented for all files, there should be a performance increase
for large models such as ScalableTestSuite.

### CODE REVIEW

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-07-28_17-09-58](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3067/).

**Comment by OpenModelica-Hudson** [comment]
Wrong hash OMCompiler f95123c66347c22382a7628d925e09ee06dd24d9 f1236e0ce2e23392a7b6601fe71bc2fd9c96f5a0

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-07-28_17-32-51](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3069/).

File: Compiler/SimCode/SimCodeMain.mo [hunk 1]
```diff
@@ -653,11 +671,19 @@ algorithm
...
+          res := System.launchParallelTasks(numThreads, codegenFuncs, runCodegenFunc);
         end if;
-
+        strs := {};
+        for tpl in res loop
+          (true,tmp) := tpl;
+          strs := List.append_reverse(tmp, strs);
+        end for;
+        strs := listReverse(strs);
+        print(stringDelimitList(strs, ", ") + "\n");
```
Comment (hkiel) [inline Compiler/SimCode/SimCodeMain.mo]: Looks like this line prints the filenames of the partial files, which is not expected output.
@sjoelund Try to comment out!

**Comment by OpenModelica-Hudson** [comment]
Wrong hash OMCompiler 3495c9392c94bd56a7406cb39da812cbf7e57448 f1236e0ce2e23392a7b6601fe71bc2fd9c96f5a0

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-09-09_09-37-06](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3131/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-09-09_09-56-43](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3132/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-09-09_10-43-32](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3133/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-09-09_11-13-11](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3134/).

**Comment by sjoelund** [comment]
Still missing: fileNamePrefix including the fmutmp search path (to generate split files within FMUs)

**Comment by hkiel** [comment]
You could leave the files unsplitted for FMU if that is easier...

**Comment by casella** [comment]
Eventually, we are planning to use FMI/CS to wrap large network simulators, so in the long run I'd like to have split files for generating FMUs as well. But that' not happening anytime soon, so maybe you could start pushing to master what @hkiel suggested and complete the FMU feature afterwards

**Comment by sjoelund** [comment]
It's actually less work to fix it for FMI than it would be to disable it for FMI. I work 4 hours/day at the moment, so I just stopped and added a remark for myself to remember what was missing :)

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2016-09-19_13-24-01](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3139/console).

**Comment by sjoelund** [comment]
So performance is slightly worse since I always generate an extra file for the initialization. So I have to decide if I should generate everything in a single file without overhead if the number of equations are low (which is then bad because around zero test cases would then actually test the new code generation)...

@adrpo, any proposal?

**Comment by casella** [comment]
How much worse is that, 1%, 10%, or 50%? In generale, I'd say an optimization should not make things worse, so I guess keeping everything in a single file would make sense for small systems.

As to testing, I think there are enough large (and sufficiently diverse) models in the ScalableTestSuite. Isn't that good enough?

You may also set a reasonable default limit size, but then set it to a lower threshold in Hudson to increase the testing coverage

My 2 cts :)

**Comment by sjoelund** [comment]
It's about 2% of overall time it seems. ScalableTestSuite is not run on every commit, although I suppose I could add a total model for it that is affected since the point is to make those models run faster...

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-09-20_08-53-36](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3140/).

**Comment by casella** [comment]
2% of a few seconds is not such a big deal...

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-09-20_09-35-11](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3141/).

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2016-09-20_10-13-21](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3142/console).

**Comment by OpenModelica-Hudson** [comment]
The job failed the coding style; for details, see [OpenModelica_TEST_PULL_REQUEST 2016-09-20_11-04-00](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3143/console). Remember to clean up the commit you did that breaks the coding style rather than piling a new commit on top of the old ones. This goes double for accidental addition of files that you need to remove.

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2016-09-20_11-06-50](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3144/).

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2016-10-13_13-34-46](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/3216/console).

**Pull request closed without merging**